### PR TITLE
fix: tell user column and q are required in nearest

### PR DESCRIPTION
**Pull request by wjones127**
This implementation just takes advantage of the default arg handling of Python.

Error message if you forget `column`:

```
TypeError: ScannerBuilder.nearest() missing 1 required positional argument: 'column'
```

Error message if you forget `column` and `q`:

```
TypeError: ScannerBuilder.nearest() missing 2 required positional arguments: 'column' and 'q'
```

### CODE REVIEW

**Comment by wjones127** [comment]
✅ I've verified that lancedb's unit tests pass with this change.

**Review by changhiskhan — COMMENTED** [review]
So if you input a bad nearest argument, after the change does it raise an exception or crash python?

**Comment by wjones127** [comment]
> So if you input a bad nearest argument, after the change does it raise an exception or crash python?

It raises a TypeError exception.

File: python/python/lance/dataset.py [hunk 1]
```diff
@@ -975,18 +976,14 @@ def with_fragments(
 
     def nearest(
         self,
-        column: Optional[str] = None,
-        q: Optional[pa.FloatingPointArray] = None,
+        column: str,
```
Comment (westonpace) [inline python/python/lance/dataset.py:979]: For some reason I thought `column` could default to "vector" but maybe that is applied elsewhere?  It's also not really an intuitive default so I wouldn't mind if we did away with it.

**Review by westonpace — APPROVED** [review]
This is fine. We might want to be consistent across all of these methods at some point.

Some, like `batch_size` (and now `nearest`) are "set-only" and others (like `batch_readahead`, `limit`, `offset`) can be "cleared" by passing in None.

We could maybe just change all of them to "set-only" at some point.  Although this is very minor.

File: python/python/lance/dataset.py (hunk 1 above)
Comment (wjones127) [reply python/python/lance/dataset.py:979]: That's applied at LanceDB level.

**Comment by wjones127** [comment]
> This is fine. We might want to be consistent across all of these methods at some point.

Yeah the builder isn't fully a public API anyways. I think most users just pass through the query arguments in `to_table()` and related methods.

**Pull request merged**